Source/Serilog.Enrichers.Span/ActivityEnricher.cs

```diff
@@ -9,6 +9,60 @@ namespace Serilog.Enrichers.Span
     /// </summary>
     public class ActivityEnricher : ILogEventEnricher
     {
+#if NET5_0
+        /// <summary>
+        /// Enrich the log event.
+        /// </summary>
+        /// <param name="logEvent">The log event to enrich.</param>
+        /// <param name="propertyFactory">Factory for creating new properties to add to the event.</param>
+        public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
+        {
+            var activity = Activity.Current;
+
+            if (activity is not null)
+            {
+                AddSpanId(logEvent, activity);
+                AddTraceId(logEvent, activity);
+                AddParentId(logEvent, activity);
+            }
+        }
+
+        private static void AddSpanId(LogEvent logEvent, Activity activity)
+        {
+            var property = activity.GetCustomProperty("Serilog.SpanId");
+            if (property is null || property is not LogEventProperty logEventProperty)
+            {
+                logEventProperty = new LogEventProperty("SpanId", new ScalarValue(activity.GetSpanId()));
+                activity.SetCustomProperty("Serilog.SpanId", logEventProperty);
+            }
+
+            logEvent.AddPropertyIfAbsent(logEventProperty);
+        }
+
+        private static void AddTraceId(LogEvent logEvent, Activity activity)
+        {
+            var property = activity.GetCustomProperty("Serilog.TraceId");
+            if (property is null || property is not LogEventProperty logEventProperty)
+            {
+                logEventProperty = new LogEventProperty("TraceId", new ScalarValue(activity.GetTraceId()));
+                activity.SetCustomProperty("Serilog.TraceId", logEventProperty);
+            }
+
+            logEvent.AddPropertyIfAbsent(logEventProperty);
+        }
+
+        private static void AddParentId(LogEvent logEvent, Activity activity)
+        {
+            var property = activity.GetCustomProperty("Serilog.ParentId");
+            if (property is null || property is not LogEventProperty logEventProperty)
+            {
+                logEventProperty = new LogEventProperty("ParentId", new ScalarValue(activity.GetParentId()));
+                activity.SetCustomProperty("Serilog.ParentId", logEventProperty);
+            }
+
+            logEvent.AddPropertyIfAbsent(logEventProperty);
+        }
+#elif NETCOREAPP3_0 || NETCOREAPP3_1
```

Comment:
Can we just use an `#else`?

Reply:
I wanted to specify the TFM, so when there's a new TFM is added to the project (net6.0) the code won't compile, until the TFM is added to its right place.

I will reverse the if and use `#else` so the new TFM uses the new implementation.

Reply:
You can use `#if NET5_0_OR_GREATER` and `#else`.

Reply:
This will make it easier to maintain when a new TFM turns up.

Reply:
Done!